text/0005-application-formalization.md

```diff
@@ -0,0 +1,191 @@
+
+**ModuleId Properties**
+| Property | Type    | Description                                                          | Notes                            |
+| -------- | ------- | -------------------------------------------------------------------- | -------------------------------- |
+| id       | String  | Identifier for the module conforming to pattern: `{name}-{version}`  | e.g. "mod-orders-storage-13.5.0" |
+| name     | String  | Name of the module                                                   | e.g. "mod-orders"                |
+| version  | String  | Version of the module                                                | e.g. "13.5.0"                    |
+| url      | String  | URL pointing to the module descriptor for this module                | e.g. "https://folio-registry.dev.folio.org/_/proxy/modules/mod-orders-storage-13.5.0" |
+
+Example:
+```
+{
+  "id": "app-gobi-0.0.1",
+  "name": "app-gobi",
+  "version": "0.0.1",
+  "description": "Application facilitating the placement of orders via GOBI and other compatible vendors",
+  "dependencies": [
+    {
+      "id": "app-orders-0.0.1",
+      "name": "app-orders",
+      "version": "^0.0.1"
+    },
+    {
+      "id": "app-organizations-0.0.1",
+      "name": "app-organizations",
+      "version": "^0.0.1"
+    }
+  ],
+  "modules": [
+    {
+      "id": "mod-gobi-2.6.0",
+      "name": "mod-gobi",
+      "version": "2.6.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/mod-gobi-2.6.0"
+    }
+  ],
+  "uiModules": [
+    {
+      "id": "folio_gobi-settings-2.0.0",
+      "name": "folio_gobi-settings",
+      "version": "2.0.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/folio_gobi-settings-2.0.0"
+    }
+  ]
+}
+```
+**N.B.** Full module descriptors and metadata have been omitted for readability reasons.
+
+## Benefits
+* Allows for expanding the bounded context
+  * By expanding the bounded context from the module to the Application, multiple modules can directly access the storage of the microservice.
+  * Benefites of this include:  
+    * Improved performance and simplified logic.  Modules can now perform the same operations with fewer API calls.
+    * Fewer modules since this eliminates the need for separate business logic and storage modules.
+* This is the first step toward breaking the monolithic release into application-specific releases
+* Results in fewer things to work with at the system operator level
+* Brings us closer to realizing the idea of having an application store/marketplace
+
+## Risks and Drawbacks
+* Folio has too many module dependencies which must be reduced
+* Transition to Applications will need to be iterative
+  * First pass is to create larger applications spanning multiple areas in order to satisfy dependencies
+  * Subsequent passes will further break up these large applications into smaller, applications focused on a single area of functionality
```

Comment:
Several TC questions were related to this iterative transition.

- How do we ensure that the initial group of applications continues to be formalized? The presentation implies inventory will be left as a mega application (p.24), how do we keep it from staying that way? 
  - Who will be responsible for deciding application boundaries, both at the interim state, and in the final state of decomposition?
  - During the time inventory is a mega application, who owns it?

Reply:
Is this applicable to the scope/general direction of the RFC? Are these questions better asked in a later stage?

Reply:
Not related to scope/general direction.  I asked them at this stage only because they complicate the risk/drawback statement I referenced above.  It would be fine to list them in unanswered questions?  Or I can ask them again in the next phase.

Reply:
Let's resolve this in scope of the PRELIMINARY REVIEW stage since it's not about scope/general direction.  I will address this in the DRAFT REFINEMENT stage.